### PR TITLE
reflected pageview PR feedback 

### PR DESCRIPTION
**Pull request by roroross**
reflected pageview PR feedback. trying to get a hang on how comments work and if i cant directly make the changes on git

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -19,7 +19,13 @@ <h1>Welcome to Rosanna's Portfolio! </h1>
       <p>Click here to get a greeting in a random language:</p>
       <button onclick="addRandomGreeting()">Hello</button>
       <div id="greeting-container"></div>
-      
+
+        <h2> Week 3 Servlet:</h2>
+        <p>Click <a href="/data">here</a> to view content generated by a servlet.</p>
+        <p>Click here to directly see content requested from /data. It should return "hello Rosanna":</p>
```
Comment (sbland-g) [inline portfolio/src/main/webapp/index.html:25]: Quoted content is not an accurate reflection of what's returned?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -63,4 +63,59 @@ function random_image() {
   // Remove the previous image.
   imageContainer.innerHTML = '';
   imageContainer.appendChild(imgElement);
+}
+
+
```
Comment (sbland-g) [inline portfolio/src/main/webapp/script.js:68]: Lots of extra white space here; please review the JavaScript style guide:
https://google.github.io/styleguide/jsguide.html#formatting-vertical-whitespace

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -63,4 +63,59 @@ function random_image() {
...
+  const responsePromise = fetch('/data');
+
+  /*
+    the responsePromise variable points to a Promise, and the code call the then() func, 
+    passing the handleresponse function in as an argument.
+    JS can reference a function using the name.
+  */
+  // When the request is complete, pass the response into handleResponse().
+  // tells JavaScript to call the handleResponse() function when the server returns a response.
+  responsePromise.then(handleResponse);
```
Comment (sbland-g) [inline portfolio/src/main/webapp/script.js:96]: What happens here if the fetch() results in an error?

File: walkthroughs/week-3-server/examples/page-view-counter/src/main/java/com/google/sps/servlets/PageViewServlet.java [hunk 4]
```diff
@@ -20,16 +20,23 @@
...
-/** Servlet that returns HTML that contains the page view count. */
-@WebServlet("/page-views")
+/** Servlet(@webservlet) that returns HTML that contains the page view count when triggered by client request. */
+
+@WebServlet("/page-views") 
 public class PageViewServlet extends HttpServlet {
 
   private int pageViews = 0;
-
+  //do get function below runs whenever a client sends a GET request to the servlet URL
```
Comment (sbland-g) [inline walkthroughs/week-3-server/examples/page-view-counter/src/main/java/com/google/sps/servlets/PageViewServlet.java:29]: When referring to a function in the comments, ensure the string matches (ie, "do get" != "doGet").

Specifically when documenting a function, consider using JSDoc formatting:
https://developers.google.com/closure/compiler/docs/js-for-compiler

But if you don't use JSDoc, please use a **consistent** block comment style:
https://google.github.io/styleguide/jsguide.html#formatting-block-comment-style

File: walkthroughs/week-3-server/examples/page-view-counter/src/main/java/com/google/sps/servlets/PageViewServlet.java [hunk 5]
```diff
@@ -20,16 +20,23 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns HTML that contains the page view count. */
-@WebServlet("/page-views")
+/** Servlet(@webservlet) that returns HTML that contains the page view count when triggered by client request. */
```
Comment (sbland-g) [inline walkthroughs/week-3-server/examples/page-view-counter/src/main/java/com/google/sps/servlets/PageViewServlet.java:23]: The /** prefix suggests this should be a JSDoc comment, but ... it's not...

File: walkthroughs/week-3-server/examples/page-view-counter/src/main/java/com/google/sps/servlets/PageViewServlet.java [hunk 6]
```diff
@@ -20,16 +20,23 @@
...
+  //aka browser seds a GET request whenever I visit a URL
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     pageViews++;
-
+    
+    /*
+    setContentType("text/html;") - specifies what type of content the client should expect
+    getWriter().println("<h1>Page Views</h1>")-  prints an <h1> tag to the response.
+    getWriter().println("<p>This page has been viewed " + pageViews + " times.</p>"); prints the page view count to the response.
```
Comment (sbland-g) [inline walkthroughs/week-3-server/examples/page-view-counter/src/main/java/com/google/sps/servlets/PageViewServlet.java:38]: Remove commented out code.

File: walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/script.js [hunk 7]
```diff
@@ -14,14 +14,27 @@
 
 /**
  * Fetches a random quote from the server and adds it to the DOM.
+ * aka JavaScript that fetches content from the server and adds it to the DOM.
  */
 function getRandomQuote() {
   console.log('Fetching a random quote.');
 
   // The fetch() function returns a Promise because the request is asynchronous.
+  /* fetch('/random-quote') sends a request to the /random-quote URL.
```
Comment (sbland-g) [inline walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/script.js:23]: Consistent block comment annotations please.

**Review by sbland-g — CHANGES_REQUESTED** [review]
Automatically generated files should not be included in the repository (ie, if you didn't deliberately make the edits, don't commit it).

The .classpath, and .settings files should be added to your git ignore configuration:
https://git-scm.com/docs/gitignore

and removed from this PR.

Update: I've injected a commit to this PR which addresses this. The rest is fine to take on as just feedback for future PRs.

**Review by sbland-g — APPROVED** [review]
LGTM; take a look at the style guide (in particular, comments and whitespace), and try incorporate the recommended style going forward.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (roroross) [reply portfolio/src/main/webapp/script.js:96]: should i have a catch for error catching?

Comment (sbland-g) [reply portfolio/src/main/webapp/script.js:96]: In general, yes, you should ensure you're doing something (at least logging) error conditions. It helps a lot when troubleshooting later if an unexpected error starts happening.

**Pull request merged**